File: new-algorithm/src/javaalgo/haulzzang/week1/boj10773.java
package javaalgo.haulzzang.week1;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class boj10773 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int K = Integer.parseInt(br.readLine());

        Stack<Integer> stack = new Stack<>();

        while (K-- > 0) {
            int num = Integer.parseInt(br.readLine());

            if (num == 0) stack.pop();
            else stack.push(num);
        }

        System.out.println(stack.stream().mapToInt(Integer::intValue).sum());
    }
}

File: new-algorithm/src/kotlinalgo/stack/boj10773.kt
package kotlinalgo.stack

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val K = Integer.parseInt(readLine())
    var sum = 0
    val stack = Stack<Int>()

    for (i in 0 until  K) {
        val num = Integer.parseInt(readLine())
        if (num != 0) stack.push(num)
        else stack.pop()
    }

    for (i in stack) {
        sum += i
    }

    println(sum)
}